### PR TITLE
[Ready Again]Refactors disabilities into lists, allowing for independent disability sources

### PR DESCRIPTION
**Pull request by XDTM**
With this you can now add and remove disabilities without worrying about other sources adding/removing them! No more "but what if the guy was already mute from genetics"?

Procs:

`add_disability(disability, source)`
`disability` is the usual disability flag. Now they're strings, and readable from view variables.
`source` is an arbitrary string that describes the source. It should be very specific for timed sources (ex. a spell that blinds for 10 seconds) to avoid interference.

`remove_disability(disability, list/sources)`
You can add multiple sources to remove. If there is no sources argument the proc will remove all sources indiscriminately.

`has_disability(disability, list/sources)`
Returns true if the target has the disabilities from at least one of the sources in the argument. If there is no sources argument, any source will count as true.

`remove_all_disabilities()`
Removes all disabilities. Should only really be used for adminheals.

BLIND, NEARSIGHT and HUSK have their own procs as they have other effects:
`become_blind(source)`, `cure_blind(list/sources)`, and so on

### CODE REVIEW

**Comment by optimumtact** [comment]
and now they're slow

**Comment by XDTM** [comment]
are we adding disabilities at such a fast rate that it's better to have conflicting disabilities though

**Comment by duncathan** [comment]
I feel like these would be better as datums than like this

**Comment by XDTM** [comment]
Wouldn't datums be even worse for performance?

Datums would probably allow for procs on gain and so on, but disabilities are mostly there to act as a static value to check on in other parts of the code, so a list is fine imo

File: code/datums/diseases/advance/symptoms/vision.dm [hunk 1]
```diff
@@ -111,16 +112,16 @@ Bonus
 		return
 	switch(A.stage)
 		if(4, 5) //basically oculine
-			if(M.disabilities & BLIND)
+			if(M.has_disability(BLIND, "damage"))
```
Comment (ninjanomnom) [inline code/datums/diseases/advance/symptoms/vision.dm]: all the string args for this proc need to be defines

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by XDTM** [comment]
Added defines for the common sources; the other ones are pretty much all one-offs, intentionally so, to prevent conflicts, and shouldn't really need to clutter the define file.

Shout out to @Xhuis for conflicting both my open PRs and making me redo everything in the mutation files ree

**Comment by ninjanomnom** [comment]
No they should still be defines, the point of it is so that if someone later expands the source it throws a compile error if they mistype rather than a hard to debug bug.

**Comment by XDTM** [comment]
Good point, i'll do that

**Comment by XDTM** [comment]
@ninjanomnom did the defines

**Comment by PKPenguin321** [comment]
does this include powers as well

**Comment by XDTM** [comment]
Codewise disabilities aren't genetic mutations, but another thing. Although a lot of genetic mutations do apply disabilities; this pr prevents healing the deafness you got from flashbangs by adding and removing the deaf mutation.

File: code/modules/mob/living/brain/brain_item.dm [hunk 2]
```diff
@@ -79,8 +79,8 @@
 		if(!brainmob.stored_dna)
 			brainmob.stored_dna = new /datum/dna/stored(brainmob)
 		C.dna.copy_dna(brainmob.stored_dna)
-		if(L.disabilities & NOCLONE)
-			brainmob.disabilities |= NOCLONE	//This is so you can't just decapitate a husked guy and clone them without needing to get a new body
+		if(L.has_disability(NOCLONE))
+			brainmob.disabilities[NOCLONE] = L.disabilities[NOCLONE]
```
Comment (AnturK) [inline code/modules/mob/living/brain/brain_item.dm]: Why is this not using the helper ?

File: code/modules/mob/status_procs.dm [hunk 3]
```diff
@@ -227,29 +227,43 @@
 /mob/proc/set_disgust(amount)
 	return
 
+/////////////////////////////////// DISABILITIES ////////////////////////////////////
+
+/mob/proc/add_disability(disability, origin)
```
Comment (AnturK) [inline code/modules/mob/status_procs.dm]: All these helpers shouldn't be here if it's living level functionality

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/brain/brain_item.dm (hunk 2 above)
Comment (XDTM) [reply code/modules/mob/living/brain/brain_item.dm]: this is copying all the sources from the mob to the brain; i could make a separate helper for it, if you want me to

File: code/modules/mob/status_procs.dm (hunk 3 above)
Comment (XDTM) [reply code/modules/mob/status_procs.dm]: It'd mean an isliving check every time you try giving a disability. Is it important?

Comment (XDTM) [reply code/modules/mob/status_procs.dm]: I mean, there's a set_disgust that just returns three lines above this

Comment (AnturK) [reply code/modules/mob/status_procs.dm]: This how you get mob_defines.dm

File: code/modules/mob/living/brain/brain_item.dm (hunk 2 above)
Comment (AnturK) [reply code/modules/mob/living/brain/brain_item.dm]: No, that's fine as is.

File: code/modules/mob/status_procs.dm (hunk 3 above)
Comment (XDTM) [reply code/modules/mob/status_procs.dm]: Irrelevant vars clutter the vv window, but mob procs add about 10 lines to a single file and save a lot more in other files in typechecking

**Pull request merged**